src/arch/linux/mapping/mapping.c

```diff
@@ -671,6 +659,20 @@ void *alloc_mapping(int cap, size_t mapsize)
   return addr;
 }
 
+void *alloc_mapping(int cap, size_t mapsize)
+{
+  Q__printf("MAPPING: alloc, cap=%s size=%#zx\n", cap, mapsize);
+  return alloc_mapping_common(cap, mapsize,
+	mmap(NULL, mapsize, PROT_NONE, MAP_ANONYMOUS | MAP_PRIVATE, -1, 0));
```

Comment:
Wouldn't it be better to pass -1 as
"target", as we always do everywhere?

Reply:
I don't think so, this is just a simple split of `alloc_mapping()` into `alloc_mapping()` and `alloc_mapping_common()`. If I pass `-1` to `alloc_mapping_common()` it'll have to check for it.

Reply:
oh I misunderstood, sure then the intermediate anon mapping is no longer necessary without huge pages.

Reply:
Also renamed to `do_alloc_mapping()`, a bit more consistent with the other function names.